python/tvm/topi/hexagon/slice_ops/max_pool2d.py

```diff
@@ -0,0 +1,193 @@
+    if iw < (ow - 1) * sw + dw * (kw - 1) + 1:
+        raise RuntimeError("Output width is too large")
+
+
+def max_pool2d_compute(A, out_shape, kernel, stride, dilation):
+    """max_pool2d compute"""
+    kh, kw = kernel
+    rh = te.reduce_axis((0, kh), name="rh")
+    rw = te.reduce_axis((0, kw), name="rw")
+    ob, oh, ow, oc = out_shape
+    if isinstance(ob, int):
+        validate_out_shape(out_shape, A.shape, kernel, stride, dilation)
+
+    sh, sw = stride
+    dh, dw = dilation
+
+    Max = te.compute(
+        out_shape,
+        lambda b, h, w, c: te.max(
+            A[b, h * sh + dh * rh, w * sw + dw * rw, c].astype(A.dtype), axis=[rh, rw]
+        ),
+        name="max",
+    )
+    return Max
+
+
+def STIR_schedule_nhwc_8h2w32c2w(outs, ins, output_layout: str, input_layout: str):
+    """Schedule for input and output layout nhwc-8h2w32c2w"""
+    func = te.create_prim_func([ins, outs])
+    s = tir.Schedule(func)
+
+    # NOTE!!! This scheduling logic is a work in progress.
+    # It is not known to ultimately result in near-optimal Hexagon performance.
+    # The schedule below strives to implement these heuristics:
+    #
+    # - Each 2048-byte chunk of the output tensor should be visited only once, if possible.
+    #
+    # - The resulting object code should use Hexagon v69's HVX SIMD units if at all possible.
+    #   (The HVX SIMD registers are 2048 bytes long, so each "chunk" of the output tensor
+    #   fits exactly in a single HVX SIMD register.)
+
+    Max = s.get_block("max")
+
+    input_transform_fn = get_layout_transform_fn(input_layout)
+    output_transform_fn = get_layout_transform_fn(output_layout)
+
+    s.transform_layout(Max, ("read", 0), input_transform_fn)
+    s.transform_layout(Max, ("write", 0), output_transform_fn)
+
+    # Restructure the loop nestings to have this overall structure:
+    # (loop over different 2048-byte output-tensor chunks) : n, ho, wo, co   }- the first level of a two-level tensor layout
+    #    (loop within one 2048-byte output-tensor chunk) : hi, wio, ci, wii  }- the second level of a two-level tensor layout
+    #        (loop over reduction axes) : rh, rw                             }- loop over multiple elements of the input tensor
+    #
+    # Note: This schedule is a work in progress.  We *expect* that it's
+    # crucially important for the loops to have this relative ordering:
+    #    n ... ho ... wo ... co ... hi ... wio ... ci ... wii
+    # because it lets us visit each of the 2048-byte output chunks precisely once.
+
+    (
+        n,
+        h,
+        w,
+        c,
+        rh,
+        rw,
+    ) = s.get_loops(Max)
+
+    # Restructure the loops from NHWC to nhwc_8h2w32c2w, with loops for 'sum's reduction
+    # axes at the very end.
+    ho, hi = s.split(h, [None, 8])
+    wo, wi = s.split(w, [None, 4])
+    wio, wii = s.split(wi, [None, 2])
+    co, ci = s.split(c, [None, 32])
+    s.reorder(n, ho, wo, co, hi, wio, ci, wii, rh, rw)
```

Comment:
From talking with @Lunderberg , I got the impression that some or all of these loop manipulations were expected to already be done by the earlier call to `transform_layout`.  

TODO(cconvey): Figure out if my understanding was wrong, or there's a bug in transform_layout.

Reply:
Yes, I have been wondering about that too. I know that transform_layout on a te.compute does the splitting as per the layout but this feature was missing from stir transform_layout and doesn't look like it has been added yet. Otherwise, it wouldn't caused failures.